### ISSUE TITLE
Add support for Python 3.12

### ISSUE THREAD

**Issue by Clemo97** [openclimatefix/pv-site-prediction#122]
## Describe the bug
[Open-Source-Quartz-Solar-Forecast](https://github.com/openclimatefix/Open-Source-Quartz-Solar-Forecast) is dependent on pv-site-prediction. Some dependencies within pv-site-prediction are out of date and not supported by Python 3.12.
## To Reproduce
1. Make sure you have Python 3.12 installed and clone [Open-Source-Quartz-Solar-Forecast](https://github.com/openclimatefix/Open-Source-Quartz-Solar-Forecast) 
2. Run `pip install quartz-solar-forecast`

![Screenshot from 2024-07-09 11-12-40](https://github.com/openclimatefix/pv-site-prediction/assets/43673340/4e701f8a-8735-45b4-98cf-2b265648d0b6)

![Screenshot from 2024-07-09 11-14-20](https://github.com/openclimatefix/pv-site-prediction/assets/43673340/ff8f8512-2a60-4e2c-9119-d86e763776ef)

**Comment by peterdudfield** [comment]
@Clemo97 did this work?

**Comment by peterdudfield** [comment]
This might need chaning - https://github.com/openclimatefix/pv-site-prediction/blob/main/pyproject.toml#L13

**Comment by zakwatts** [comment]
pandas-stubs should now be able to be updated as well: https://github.com/pandas-dev/pandas-stubs/issues/625

**Comment by Clemo97** [comment]
> @Clemo97 did this work?

It did but unfortunately my computer died before I could push my changes😭 😭, hence why I never made a PR. All that's required is running the "poetry update" commands check[ poetry docs](https://python-poetry.org/docs/cli#update) for that  which will fail on some dependencies which you then have to manually adjust their version to one that is compatible to Python 3.12.
Pandas-stubs is just the first dependency that's incompatible but there are more, I found like 3 others.

**Comment by zakwatts** [comment]
Ahhh I do hate it when that happens! Okay, I'll have a look into this again now. Thanks for the effort you have put in!

**Comment by jamescrowley** [comment]
Just hit this myself - any update re getting https://github.com/openclimatefix/pv-site-prediction/pull/124 merged? Happy to support if needed.

**Comment by Clemo97** [comment]
no update yet, this issue is actually bigger than it looks.
I choose to downgrade my python version to 3.11.
You can look in to it if you like

**Comment by gilbertgong** [comment]
Yes I was in the middle of editing/updating my comment to say I wasn't sure what the alternatives are to essentially requiring model retraining and making old models binary incompatible, along with clearly documenting this and pointing to older versions for older model support.  I looked at some possible solution with a point release that could do an ONNX conversion of a pkl model, but not sure if that's even viable or desirable.
